src/app/GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -561,10 +561,20 @@ protected override void OnActivated(EventArgs e)
                         new WindowsThumbnailToolbarButton(_closeAll.Text, Images.DeleteFile, (s, e) => NativeMethods.PostMessageW(NativeMethods.HWND_BROADCAST, _closeAllMessage))));
             }
 
+            _windowsJumpListManager.EnableThumbnailToolbar(_dashboard?.Visible is not true && Module.IsValidGitWorkingDir());
```

Comment:
Add a comment here or to the commit description, this is another fix (that I could not relate to the other)

Reply:
Return from modal dialog does not work without this.
But yes, it fixes the bug that the buttons must be disabled for the Dashboard (refer to PR description and to updated commit message).